diff --git a/configure.ac b/configure.ac
index de2bcdb..679587c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -401,13 +401,7 @@ AC_ARG_WITH(opencl,
 
 if test $with_opencl != "no" ; then {
 	AC_CHECK_HEADER(CL/opencl.h,,AC_MSG_ERROR([ ** You need to install opencl-headers package.]))
-	PKG_CHECK_MODULES(LIBCL, OpenCL,[
-		CONFIG_DEPS="$CONFIG_DEPS OpenCL"
 		with_opencl="yes"
-	],[
-		AC_MSG_ERROR([ ** You need to install OpenCL.])
-		with_opencl="no"
-	])
 	
 } ; fi
 AM_CONDITIONAL(WITH_OPENCL, test $with_opencl = yes)
