diff --git a/src/amd/llvm/ac_llvm_helper.cpp b/src/amd/llvm/ac_llvm_helper.cpp
index 0316e1d..f7d25e3 100644
--- a/src/amd/llvm/ac_llvm_helper.cpp
+++ b/src/amd/llvm/ac_llvm_helper.cpp
@@ -309,9 +309,7 @@ LLVMValueRef ac_build_atomic_rmw(struct ac_llvm_context *ctx, LLVMAtomicRMWBinOp
    unsigned SSID = llvm::unwrap(ctx->context)->getOrInsertSyncScopeID(sync_scope);
    return llvm::wrap(llvm::unwrap(ctx->builder)
                         ->CreateAtomicRMW(binop, llvm::unwrap(ptr), llvm::unwrap(val),
-#if LLVM_VERSION_MAJOR >= 13
-                                          llvm::MaybeAlign(0),
-#endif
+
                                           llvm::AtomicOrdering::SequentiallyConsistent, SSID));
 }
 
@@ -322,9 +320,7 @@ LLVMValueRef ac_build_atomic_cmp_xchg(struct ac_llvm_context *ctx, LLVMValueRef
    return llvm::wrap(llvm::unwrap(ctx->builder)
                         ->CreateAtomicCmpXchg(llvm::unwrap(ptr), llvm::unwrap(cmp),
                                               llvm::unwrap(val),
-#if LLVM_VERSION_MAJOR >= 13
-                                              llvm::MaybeAlign(0),
-#endif
+
                                               llvm::AtomicOrdering::SequentiallyConsistent,
                                               llvm::AtomicOrdering::SequentiallyConsistent, SSID));
 }
