diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3105458..9dd5fa3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -446,7 +446,7 @@ else()
   option(WITH_CYCLES_DEVICE_HIP        "Enable Cycles AMD HIP support" OFF)
 endif()
 option(WITH_CYCLES_HIP_BINARIES      "Build Cycles AMD HIP binaries" OFF)
-set(CYCLES_HIP_BINARIES_ARCH gfx1010 gfx1011 gfx1012 gfx1030 gfx1031 gfx1032 gfx1034 CACHE STRING "AMD HIP architectures to build binaries for")
+set(CYCLES_HIP_BINARIES_ARCH gfx803 CACHE STRING "AMD HIP architectures to build binaries for")
 mark_as_advanced(WITH_CYCLES_DEVICE_HIP)
 mark_as_advanced(CYCLES_HIP_BINARIES_ARCH)
 