diff --git a/extern/hipew/src/hipew.c b/extern/hipew/src/hipew.c
index 010c361..747dd78 100644
--- a/extern/hipew/src/hipew.c
+++ b/extern/hipew/src/hipew.c
@@ -253,7 +253,7 @@ static int hipewHipInit(void) {
   /* Default installation path. */
   const char *hip_paths[] = {"", NULL};
 #else
-  const char *hip_paths[] = {"/opt/rocm/hip/lib/libamdhip64.so", NULL};
+  const char *hip_paths[] = {"/usr/lib/hip/lib/libamdhip64.so.4", NULL};
 #endif
   static int initialized = 0;
   static int result = 0;
