diff --git a/kiwixbuild/platforms/native.py b/kiwixbuild/platforms/native.py
index 1a0c2c6..bb0c5f5 100644
--- a/kiwixbuild/platforms/native.py
+++ b/kiwixbuild/platforms/native.py
@@ -17,7 +17,7 @@ class NativePlatformInfo(PlatformInfo):
 class NativeDyn(NativePlatformInfo):
     name = 'native_dyn'
     static = False
-    compatible_hosts = ['fedora', 'debian', 'Darwin']
+    compatible_hosts = ['fedora', 'debian', 'Darwin', 'gentoo']
 
 class NativeStatic(NativePlatformInfo):
     name = 'native_static'
