--- a/CMakeLists.txt	2021-03-28 18:02:34.755857717 +0200
+++ b/CMakeLists.txt	2021-03-28 18:04:54.217854203 +0200
@@ -208,15 +208,15 @@
 endif()
 message(STATUS "HIP Platform: " ${HIP_PLATFORM})
 
-if(HIP_PLATFORM STREQUAL "nvidia")
-    set(HIP_RUNTIME "cuda"  CACHE STRING "HIP Runtime")
-    set(HIP_COMPILER "nvcc" CACHE STRING "HIP Compiler")
-elseif(HIP_PLATFORM STREQUAL "amd")
-    set(HIP_RUNTIME "rocclr" CACHE STRING "HIP Runtime")
-    set(HIP_COMPILER "clang" CACHE STRING "HIP Compiler")
-else()
-    message(FATAL_ERROR "Unexpected HIP_PLATFORM: " ${HIP_PLATFORM})
-endif()
+if(HIP_PLATFORM STREQUAL "nvidia")
+    set(HIP_RUNTIME "cuda"  CACHE STRING "HIP Runtime")
+    set(HIP_COMPILER "nvcc" CACHE STRING "HIP Compiler")
+elseif(HIP_PLATFORM STREQUAL "amd")
+    set(HIP_RUNTIME "rocclr" CACHE STRING "HIP Runtime")
+    set(HIP_COMPILER "clang" CACHE STRING "HIP Compiler")
+else()
+    message(FATAL_ERROR "Unexpected HIP_PLATFORM: " ${HIP_PLATFORM})
+endif()
 
 message(STATUS "HIP Runtime: " ${HIP_RUNTIME})
 message(STATUS "HIP Compiler: " ${HIP_COMPILER})
